src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect } from 'react';
+import { Range } from 'react-range';
+
+import useTheme from '../../hooks/useTheme';
+import { TestProps } from '../../utils/types';
+import TextField from '../TextField';
+import SliderMark from './components/SliderMark';
+import SliderThumb from './components/SliderThumb';
+import SliderTrack from './components/SliderTrack';
+import { Container, InputContainer, InputsContainer } from './Slider.style';
+
+interface Props {
+  /** Determines the position of the rendered thumbs. Defaults to [0, 100] */
+  values: number[];
+  /** Determines if the Slider is disabled or not */
+  disabled?: boolean;
+  /** Determines whether the Slider is a Range or a Selector */
+  isSelector?: boolean;
+  /** Determines if the Slider will show increments along its track */
+  hasIncrements?: boolean;
+  /** Use this to change the slider's initial values (e.g. [80, 100]) */
+  initialValues?: number[];
+  /** Called when a thumb is moved, provides new values */
+  onChange: (values: number[]) => void;
+  /** Called when a change is finished (mouse/touch up, or keyup), provides current values.
+   * Use this event when you have to make for example ajax request with new values. */
+  onFinalChange?: (values: number[]) => void;
+}
+
+const STEP = 1;
+export const STEP_WITH_INCREMENTS = 20;
+export const MIN = 0;
+export const MAX = 100;
+
+const Slider: React.FC<Props & TestProps> = ({
+  values,
+  initialValues = [MIN, MAX],
+  onChange,
+  onFinalChange,
+  isSelector = false,
+  hasIncrements = false,
+  disabled = false,
+  dataTestPrefixId,
+}) => {
+  const theme = useTheme();
+
+  useEffect(() => {
+    if (isSelector && values.length > 1) {
+      throw new Error(
+        'The Selector type Slider can only accept one initial value in the array, since it only has one thumb. Try changing the `initialValues` prop.'
+      );
+    }
+  }, [values, isSelector]);
+
+  const sanitizeValues = (value: number) => {
+    if (value < MIN) {
+      return 0;
+    }
+
+    if (value > MAX) {
+      return 100;
+    }
+
+    return value;
+  };
+
+  return (
+    <Container data-testid={`${dataTestPrefixId ?? ''}slider_component`}>
+      <Range
+        step={hasIncrements ? STEP_WITH_INCREMENTS : STEP}
+        min={MIN}
+        max={MAX}
+        disabled={disabled}
+        values={values}
+        onChange={onChange}
+        onFinalChange={onFinalChange}
+        renderMark={({ props, index }) =>
```

Comment:
With regards to the comment below, I personally always prefer writing a proper function body, instead of the return shorthand, because it actually allows you to do function things inside the function body, and helps with debugging too. 

I am just throwing this here, not requesting a change (even though if you decide to implement the change suggested in the  comment below, you will have to make this change here as well :hankey: )